src/fsharp/TastOps.fs

```diff
@@ -1259,8 +1259,9 @@ type ValHash<'T> =
 
     member ht.TryFind (v:Val) = 
         let (ValHash t) = ht
-        let i = v.Stamp
-        if t.ContainsKey(i) then Some(t.[i]) else None
+        match t.TryGetValue v.Stamp with
```

Comment:
Have you tried this under some performance test?

I know this sounds stupid, but when I was looking at similar stuff and did simple testing under FSI, using similar sized dictionaries than code I was looking at, the former code actually performed better, which was kind of baffling.